tex/gregoriotex-syllable.tex

```diff
@@ -556,7 +556,7 @@
   % took me years to find it!
   \gre@debugmsg{bolshift}{Current glyph: \the\gre@attr@glyph@id}%
   \ifgre@bolshiftsenabled%
-    \gre@calculate@bolshift{\gre@dimen@begindifference}%
+    \gre@calculate@bolshift{\gre@dimen@begindifference}{\gre@firstisalteration}% TODO: compute firstisalteration
```

Comment:
I don't necessarily disagree with you philosophically, it's just that TeX is designed in such a way that variables are global.  Hence when programming in TeX I tend towards the more implicit style.